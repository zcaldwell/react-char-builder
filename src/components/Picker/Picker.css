.picker {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  height: 600px;
}

.form-control {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.tracker {
  font-style: oblique;
}
