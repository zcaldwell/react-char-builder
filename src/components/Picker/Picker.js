import React from 'react';

export default function Picker({ head, setHead, body, setBody, legs, setLegs, setCatchPhrase }) {
  return (
    <div className="picker">
      <div className="form-control">
        <label>Head</label>
        <select value={head} onChange={(e) => setHead(e.target.value)}>
          <option value="bird-head">Bird Head</option>
          <option value="dog-head">Dog Head</option>
          <option value="duck-head">Duck Head</option>
          <option value="horse-head">Horse Head</option>
        </select>
        <label>Body</label>
        <select value={body} onChange={(e) => setBody(e.target.value)}>
          <option value="blue-middle">Blue Body</option>
          <option value="dress-middle">Dress Body</option>
          <option value="pink-middle">Pink Body</option>
          <option value="red-middle">Red Body</option>
        </select>
        <label>Legs</label>
        <select value={legs} onChange={(e) => setLegs(e.target.value)}>
          <option value="dog-pants">Dog Pants</option>
          <option value="blue-pants">Blue Pants</option>
          <option value="white-pants">White Pants</option>
          <option value="leg-pants">Leg Pants</option>
        </select>
      </div>
      <div className="form-control">
        <label>Enter A Catch Phrase!</label>
        <input type="text" onChange={(e) => setCatchPhrase(e.target.value)} />
      </div>
    </div>
  );
}
