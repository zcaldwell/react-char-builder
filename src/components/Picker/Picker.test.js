import { render, screen } from '@testing-library/react';
import Picker from './Picker';

test('renders the Picker panel', () => {
  const container = render(
    <Picker
      head="dog-head"
      setHead="dog-head"
      setHeadcount="0"
      body="dress-body"
      setBody="dress-body"
      setBodyCount="0"
      legs="dog-pants"
      setLegs="dog-pants"
      setLegsCount="0"
      catchPhrase="ok"
      setCatchPhrase="ok"
      setCatchList="[]"
    />
  );
  expect(container).toMatchSnapshot();
});
