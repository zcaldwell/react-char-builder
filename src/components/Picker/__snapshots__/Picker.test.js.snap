// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Picker panel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="picker"
      >
        <div
          class="form-control"
        >
          <label>
            Head
          </label>
          <select>
            <option
              value="bird-head"
            >
              Bird Head
            </option>
            <option
              value="dog-head"
            >
              Dog Head
            </option>
            <option
              value="duck-head"
            >
              Duck Head
            </option>
            <option
              value="horse-head"
            >
              Horse Head
            </option>
          </select>
          <label>
            Body
          </label>
          <select>
            <option
              value="blue-middle"
            >
              Blue Body
            </option>
            <option
              value="dress-middle"
            >
              Dress Body
            </option>
            <option
              value="pink-middle"
            >
              Pink Body
            </option>
            <option
              value="red-middle"
            >
              Red Body
            </option>
          </select>
          <label>
            Legs
          </label>
          <select>
            <option
              value="dog-pants"
            >
              Dog Pants
            </option>
            <option
              value="blue-pants"
            >
              Blue Pants
            </option>
            <option
              value="white-pants"
            >
              White Pants
            </option>
            <option
              value="leg-pants"
            >
              Leg Pants
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <label>
            Enter A Catch Phrase!
          </label>
          <input
            type="text"
            value="ok"
          />
          <button>
            Add
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="picker"
    >
      <div
        class="form-control"
      >
        <label>
          Head
        </label>
        <select>
          <option
            value="bird-head"
          >
            Bird Head
          </option>
          <option
            value="dog-head"
          >
            Dog Head
          </option>
          <option
            value="duck-head"
          >
            Duck Head
          </option>
          <option
            value="horse-head"
          >
            Horse Head
          </option>
        </select>
        <label>
          Body
        </label>
        <select>
          <option
            value="blue-middle"
          >
            Blue Body
          </option>
          <option
            value="dress-middle"
          >
            Dress Body
          </option>
          <option
            value="pink-middle"
          >
            Pink Body
          </option>
          <option
            value="red-middle"
          >
            Red Body
          </option>
        </select>
        <label>
          Legs
        </label>
        <select>
          <option
            value="dog-pants"
          >
            Dog Pants
          </option>
          <option
            value="blue-pants"
          >
            Blue Pants
          </option>
          <option
            value="white-pants"
          >
            White Pants
          </option>
          <option
            value="leg-pants"
          >
            Leg Pants
          </option>
        </select>
      </div>
      <div
        class="form-control"
      >
        <label>
          Enter A Catch Phrase!
        </label>
        <input
          type="text"
          value="ok"
        />
        <button>
          Add
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
